### DISCUSSION TITLE
CommunityToolkit.MVVM example

### DISCUSSION THREAD

**Discussion by johnmbaughman** [tui-cs/Terminal.Gui#3535] [Ideas]
Just for curiosity sake, I created an example using the [`CommunityToolkit.Mvvm` framework](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/). I followed the ReactiveUI example and it works identical to that. There was some refactoring done to make the view initialization similar to the TUI designer, just for fun.

Is this something that could be added to the project or even wanted? If so, how would I handle the PR?

It was fun to experiment and learn with this really cool framework and I'd like to share the idea with others. 

https://github.com/johnmbaughman/Terminal.Gui/tree/v2_communitytoolkit

-John

**Comment by tig** [comment]
I love it!

I can't tell if "There was some refactoring done to make the view initialization similar to the TUI designer" means you changed core code or just the react sample. Can you clarify?

We can either add this as a sample in the Terminal.Gui repo or as a showcase. 

Feel free to submit a PR against v2_develop if you think it would be  good sample!

**Reply by BDisp** [reply]
He said that followed the `ReactiveUI` example, which is a good chose to achieve the same target with different tool kits.
@johnmbaughman don't forget to add `Application.Top.Dispose ();` before `Application.Shutdown ();` otherwise an assertion will be throw at debug mode. The `ReactiveUI` also throw the assertion fail as well in debug mode. Thanks.

**Reply by BDisp** [reply]
What I recommend is to create a `README.md` file as the `ReactiveUI` example explaining how `CommunityToolkit.Mvvm` framework works with `Terminal.Gui`. Thanks.

**Reply by johnmbaughman** [reply]
> I can't tell if "There was some refactoring done to make the view initialization similar to the TUI designer" means you changed core code or just the react sample. Can you clarify?

Sorry, it was late and I was tired but wanted to post this. I meant the result of using the designer. I created the equivalent `*.designer.cs` file with the view partial class just to follow a similar flow. Silly, but it worked to keep the MVVM related code out of the basic view design code.

I'll create the `README.md` file and fix up the `.sln` file since Visual Studio did some strange things with it. Don't want to change more than necessary to add this project in. And obviously, I'll add the dispose call, and I'll add it to the ReactiveUI one as a bug fix. 

As for the PR, I'll add feature request for the template info. 

Thanks! 

-John
